sepal_ui/planetapi/planet_view.py

```diff
@@ -0,0 +1,111 @@
+from sepal_ui.planetapi import PlanetModel
+import sepal_ui.scripts.utils as su
+import sepal_ui.sepalwidgets as sw
+
+
+class PlanetView(sw.Layout):
+    """Stand-alone interface to capture planet lab credentials, validate its  subscription and
+    connect to the client stored in the model.
+
+    Args:
+        bnt (sw.Btn, optional): Button to trigger the validation process in the associated model.
+        alert (sw.Alert, v.Alert, optional): Alert component to display end-user action results.
+        planet_model (sepal_ui.planetlab.PlanetModel): backend model to manipulate interface actions.
+
+    """
+
+    def __init__(self, *args, btn=None, alert=None, planet_model=None, **kwargs):
+
+        self.class_ = "d-block flex-wrap"
+
+        super().__init__(*args, **kwargs)
+
+        self.planet_model = planet_model if planet_model else PlanetModel()
+        self.btn = btn if btn else sw.Btn("Validate", small=True, class_="mr-1")
+        self.alert = alert if alert else sw.Alert()
+
+        self.w_username = sw.TextField(
+            label="Planet username", class_="mr-2", v_model=""
+        )
+        self.w_password = sw.PasswordField(label="Planet password")
+        self.w_key = sw.PasswordField(label="Planet API key", v_model="").hide()
+
+        states = {
+            False: ("Not connected", color.error),
+            True: ("Connected", color.success),
+        }
+
+        self.w_state = sw.StateIcon(self.planet_model, "active", states)
+
+        self.w_method = v.Select(
+            label="Login method",
+            class_="mr-2",
+            v_model="credentials",
+            items=[
+                {"value": "api_key", "text": "API Key"},
+                {"value": "credentials", "text": "Credentials"},
+            ],
+        )
+
+        w_validation = v.Flex(
+            style_="flex-grow: 0 !important;",
+            children=[self.btn, self.w_state],
+            class_="pr-1 flex-nowrap",
+        )
+        self.children = [
+            self.w_method,
+            sw.Layout(
+                class_="align-center",
+                children=[
+                    self.w_username,
+                    self.w_password,
+                    self.w_key,
+                ],
+            ),
+        ]
+
+        if not btn:
+            self.children[-1].set_children(w_validation, "last")
+
+        if not alert:
+            self.set_children(self.alert, "last")
+
+        self.w_method.observe(self._swap_inputs, "v_model")
+        self.btn.on_event("click", self.validate)
+
+    def reset(self):
+        """Empty credentials fields"""
+
+        self.w_username.v_model = None
+        self.w_password.v_model = None
+        self.w_key.v_model = None
+
+    def _swap_inputs(self, change):
+        """Swap between credentials and api key inputs"""
+
+        self.planet_model.init_client(None)
+        self.alert.reset()
+        self.reset()
+
+        if change["new"] == "api_key":
```

Comment:
from what I understand if it's a swap it means that I cannot select the same value twice. what about instead of a if statement we use the `toggle_viz()` method from SepalWidget to change the 3 components status ?